### ISSUE TITLE
Style loader for sass / scss does not work

### ISSUE THREAD

**Issue by YahngSungho** [neutrinojs/neutrino#1066]
In my opinion, I've tried all the methods I've found to use the sass / scss loader. But it did not work in the end.

Someone please check...

I created an example repo:
https://github.com/YahngSungho/dfsfdsfsfdsfwfsd/tree/help_style_loader

I installed:
https://www.npmjs.com/package/neutrino-middleware-sass
https://www.npmjs.com/package/neutrino-preset-sass

My neutrino.js:
https://github.com/YahngSungho/dfsfdsfsfdsfwfsd/blob/help_style_loader/.neutrinorc.js

What I tried:
https://github.com/YahngSungho/dfsfdsfsfdsfwfsd/blob/help_style_loader/src/components/style-loader-test.story.js

What I saw:
![image](https://user-images.githubusercontent.com/29212791/44953146-6e7bc780-aecb-11e8-921c-2614c41e6d53.png)

**Comment by edmorley** [comment]
Hi!

Both of those packages are third-party and don't look actively maintained. For example `neutrino-preset-sass` has a peer dependency on the fairly old Neutrino 4:
https://github.com/kopacki/neutrino-dev/blob/15aba74502c88176eeb3ccb2a9073ae570f55be8/packages/neutrino-preset-sass/package.json#L21

Similarly, `neutrino-middleware-sass` has a peer dependency on Neutrino 6:
https://github.com/markelog/neutrino-middleware-sass/blob/a5826373417b9163fd8a7e1ac069365e4a4f2689/package.json#L28

Generally when the peer dependency versions do not match, it's not guaranteed that things will work. As such, it's worth watching the npm/yarn install log, since they will warn about version mismatches like this :-)

Thankfully for Neutrino 8, `@neutrinojs/style-loader` (which is used by the web/react/... presets) now supports passing in custom loaders directly, without the need for additional presets. There is an example of how to do this in the docs:
https://neutrinojs.org/packages/style-loader/#custom-loaders

Could you try removing the third party presets and following the steps in the Neutrino docs, and let us know how you get on?

**Comment by edmorley** [comment]
eg:

```js
module.exports = {
  use: [
    ['@neutrinojs/react', {
      // These options get passed down to `@neutrinojs/style-loader`
      style: {
        // ...
      }
    }]
  ]
};
```

**Comment by YahngSungho** [comment]
Thank you for your reply. I removed those two third-parties and tried again. But it still did not work.

My neutrino.js was:

```
[
      '@neutrinojs/react',
      {
        html: {
          title: 'App'
        }
        ,
        style: {
          test: /\.(css|sass|scss)$/,
          moduleTest: /\.module\.(css|sass|scss)$/,
          loaders: [
            { loader: require.resolve('sass-loader'), useId: 'sass' },
            {
              loader: 'postcss-loader',
              options: {
                plugins: [require('autoprefixer')]
              }
            }

          ]
        }

      }
    ]
```

In addition to this, the way I've tried is shown in my neutrino.js.

https://github.com/YahngSungho/dfsfdsfsfdsfwfsd/blob/help_style_loader/.neutrinorc.js

**Comment by edmorley** [comment]
I've checked out the repository linked above (very useful to have it - thank you!). I had to modify the `.neutrinorc.js` slightly to get it to work, since Neutrino was looking for `src/index.js` that didn't exist on that branch (I used the `mains` option to point at the component directly instead).

There appears to be three separate issues:

1) The `neutrino-preset-emotion` preset is listed before the main "build" preset (`@neutrino/react`), which causes `TypeError: Cannot read property 'plugins' of undefined`. Since it modifies the `compile` rule, it needs to be moved after the react preset.

2) After fixing that, I got:
```
./src/components/sassFile.sass
Module build failed: ModuleBuildError: Module build failed:
.styleTest1 {
  background-color: firebrick;
}
            ^
      Invalid CSS after ".styleTest1 {": expected "}", was "{"
      in C:\Users\Ed\src\dfsfdsfsfdsfwfsd\src\components\sassFile.sass (line 1, column 14)
```

Looking at the [SASS docs](https://sass-lang.com/guide#topic-3-SCSS), it seems that `sassFile.sass` is using the wrong syntax (SCSS instead of SASS), which is why it's giving that error message. Either rename the file to `.scss` or leave as `.sass` and remove the brackets and semicolon - ie: change to this for it to be valid SASS markup:

```sass
.styleTest1
  background-color: firebrick
```

3) Not sure if this issue affected you, but I spotted that our example in the docs had the order for `sass-loader` and `postcss-loader` inverted compared to what the [postcss-loader docs](https://github.com/postcss/postcss-loader#config-cascade) say:

> You can use it standalone or in conjunction with css-loader (recommended). Use it after css-loader and style-loader, but before other preprocessor loaders like e.g sass|less|stylus-loader, if you use any.

I'll open a PR to fix the docs for (3).

**Comment by YahngSungho** [comment]
Thank you for your detailed and kind answers.
I think I've done everything you asked me to do. But nevertheless this doesn't work. The same error message still appears.
![storybook - chrome 2018-09-06 20 36 36](https://user-images.githubusercontent.com/29212791/45155285-68605080-b215-11e8-9e73-7eaf75e73f9f.png)

Can you check it one more time? The part I changed can be found in the repo:
https://github.com/YahngSungho/dfsfdsfsfdsfwfsd/tree/help_style_loader

Especially in neutrino.js:
https://github.com/YahngSungho/dfsfdsfsfdsfwfsd/blob/help_style_loader/.neutrinorc.js

**Comment by YahngSungho** [comment]
I've not solved this problem yet. Does anyone know how to solve it, please?

**Comment by YahngSungho** [comment]
I am grateful for the answers in this detailed and well-documented manner.
I've fixed all the parts you pointed out. And when I ran 'neutrino start', I got 'Build completed' from the terminal as you said.

However, when I run it using storybook, It still cannot parse sass / scss. Could not I ever get the files loaded correctly when I use the storybook?

It's a repo that modified the points you pointed out:
https://github.com/YahngSungho/dfsfdsfsfdsfwfsd/tree/help_style_loader
